File: Java/Fibonacci Number.java
// 0ms, 100.00%; 36.6MB, 5.51%
class Solution {
    public int fib(int N) {
        if (N < 2) {
            return N;
        }
        int[] a = new int[N + 1];
        a[0] = 0;
        a[1] = 1;
        for (int i = 2; i <= N; ++i) {
            a[i] = a[i - 1] + a[i - 2];
        }
        return a[N];
    }
}

File: Go/Fibonacci Number.go
// 0ms, 100.00%; 1.9MB, 100.00%
func fib(N int) int {
	if N == 0 {
		return 0
	}
	a, b := 0, 1
	for i := 2; i <= N; i++ {
		a, b = b, a+b
	}
	return b
}
